keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -584,6 +586,142 @@ would allow us to delete the pod and trying again with a new one, but is not don
 at the moment because admission checks cannot be retried if a check finds
 a transient problem.
 
+#### Partitionable devices
+
+In addition to devices, a `ResourceSlice` can also embed a list of
+`SharedCapacity` objects. Each `SharedCapacity` object represents some amount
+of shared "capacity" that can be consumed by one or more devices listed in the
+slice. No individual device can consume more than the total capacity available
+in a given `SharedCapacity` object, but that is the only restriction. In other
```

Comment:
There is really no restriction on the *sum* of the consumed capacity?

For example, this here says "I have 40Gi":
```yaml
 sharedCapacity:
  - name: gpu-0-shared-memory
    capacity: 40Gi
  - name: gpu-0-memory-block-0
    capacity: 1
  - name: gpu-0-memory-block-1
    capacity: 1
  - name: gpu-0-memory-block-2
    capacity: 1
  - name: gpu-0-memory-block-3
    capacity: 1
```

With the definition above, it would be possible to allocate 4 devices which each consume one memory block and 40Gi each.

That cannot be right. It wouldn't even work for the memory blocks, because consuming `gpu-0-memory-block-0` with 1 would not prevent some other device from consuming the same.

I think this is meant:
```suggestion
slice. The sum of consumed capacity may never exceed the total capacity available
in a given `SharedCapacity` object, but that is the only restriction. In other
```

Reply:
yes, correct pohly

Reply:
That is not correct. What I write originally is what I meant.

Reply:
I am writing this from the perspective of how you build up the device objects in the ResourceSlice itself, not the how the scheduler will track it when allocating them.

Reply:
You need to document how the scheduler is meant to handle this information. There has to be some definition of a mathematical invariant that has to be enforced by the scheduler.

"Overlapping" is not a mathematical definition and from the text is is not clear how the scheduler would detect "not overlapping".

Reply:
I have updated the text at the bottom of the section with more details.

Reply:
The text below makes sense to me. The "that is the only restriction" statement above and the "can consume overlapping portions" below still don't. Can you perhaps move the precise description up directly after "can be consumed by one or more devices listed in the
slice."?

The rest then can become an illustration of the concept.

Reply:
Attempting to reword this part now...

Reply:
Rearranged and reworded. PTAL